templates/conf/log4j.properties.erb

```diff
@@ -65,3 +65,9 @@ log4j.appender.TRACEFILE.File=${zookeeper.tracelog.dir}/${zookeeper.tracelog.fil
 log4j.appender.TRACEFILE.layout=org.apache.log4j.PatternLayout
 ### Notice we are including log4j's NDC here (%x)
 log4j.appender.TRACEFILE.layout.ConversionPattern=%d{ISO8601} [myid:%X{myid}] - %-5p [%t:%C{1}@%L][%x] - %m%n
+<% @extra_appenders.each_pair do |name, data| -%>
```

Comment:
It might be a good idea to add a `.sort` here, in order to avoid unnecessary changes each puppet run.